nodes/basic-pow/src/service.rs

```diff
@@ -193,6 +196,33 @@ pub fn new_full(config: Configuration) -> Result<TaskManager, ServiceError> {
 		task_manager
 			.spawn_essential_handle()
 			.spawn_blocking("pow", worker_task);
+		
+		// Start Mining
+		let mut numb: u8 = 0;
+		thread::spawn(move || {
+			loop {
+				let worker = _worker.clone();
+				let metadata = worker.lock().metadata();
+				if let Some(metadata) = metadata {
+					let nonce = H256::from_slice(Sha3_256::digest(&[numb]).as_slice());
```

Comment:
Actually I am not familiar with writing consensus, so I am also learning here. A few questions here:

Why is `nonce` a hash, instead of simply an incrementing number? I see that numb pretty much act as an incrementing nonce. But then this number should probably be of type `U256` instead of `u8`.

Reply:
Thanks @jimmychu0807 for the review!

The `nonce` as a Hash is mimicking the Kulupu implementation that uses the RandomX algorithm, as one of it's security goals that the Miners should not be able to identify "weaker" authorised parameters (such as specific nonce values) that make the PoW computation significantly more efficient, thus giving an advantage to a miner who would know such parameters, and in Kulupu implementation the `nonce` is actully the Hash of a random value 
`let nonce = H256::random_using(&mut rng);`
In this implementation it doesn't really need to be a Hash at all, I think an `U256` would be a better fit, as we really do increment the `nonce` by `1` every unsuccessful attempt.
the `u8` was used cuz i didn't change anything in the rest of the code, where the `nonce` is a Hash, and the `Sha3_256::digest()` takes a `&[u8]` as input, so i found it a faster way.
Bottom line using the `U256` for the nonce would eliminate the need for the `numb` variable. It was just quick and dirty implementation to make it work as discussed in https://github.com/substrate-developer-hub/recipes/issues/432 
I updated the code to make it uses the type `U256` for the nonce.